### DISCUSSION TITLE
How to use the built result?

### DISCUSSION THREAD

**Discussion by neoedmund** [eclipse-jdt/eclipse.jdt.core#3193] [Q&A]
I want to build and use JDT with Eclipse IDE from source, not from the pre-built official sites.

I'v check out  the git repo source and build with something like `mvn -P build-individual-bundles package` .
and it generated some jar like `org.eclipse.jdt.core-3.40.0-SNAPSHOT.jar`

I just installed a eclipse Platform without any software , how could I use them to add into eclipse IDE?

I tried 'Install New Software' , it ask for zip/jar , but I cannot find built one can be used.

There is no manual found , so I am here to ask. Help please.

**Comment by stephan-herrmann** [comment]
If you have worked on JDT source code within Eclipse, then for temporary use the easiest would be to just launch a runtime workbench from that development workspace.

If you want something more permanent, then "Install New Software" is indeed the normal way, but for this you need more than just one plug-in jar: a feature and meta-data.

For the feature you could use https://github.com/eclipse-jdt/eclipse.jdt/tree/master/org.eclipse.jdt-feature (which then contains all of JDT, not just jdt.core), or handcraft your own (it's actually not complex).

As for creating the update site / aka p2 repository with metadata suitable for "Install New Software" I don't even think there's a prepared way to create this just for JDT (others may correct me if I'm wrong).

But again there is support right from the IDE: in the workspace where you have  plug-in project and feature project checked out and built, you can use an Export wizard to export the feature which will then create the full update site. I do this regularly, usually without any problems.

**Reply by akurtakov** [reply]
After all of PDE has been merged into single repo the master branch build has been modified to publish latest bits from its last build to https://download.eclipse.org/pde/builds/master/ and everyone can do it locally too (https://github.com/eclipse-pde/eclipse.pde/blob/b013114e227575891d20c990cc3068359e5f8d32/pom.xml#L92). 
As long as JDT is multiple repos I am out of ideas how to build and provide such a repository.

**Reply by stephan-herrmann** [reply]
> After all of PDE has been merged into single repo the master branch build has been modified to publish latest bits from its last build to https://download.eclipse.org/pde/builds/master/ 

Interesting!

As said though, I'm perfectly happy with interactively exporting features from the IDE  -- I don't even need an internet connection for that ... :)

**Reply by neoedmund** [reply]
Thank you!
Could you show me some script?
I checked out  `eclipse-jdt/eclipse.jdt` and built with `mvn package`
What location should I use as the `software site`?

**Comment by neoedmund** [comment]
I found out Eclipse even not use the version number in `feature.xml`.
But it worked that overwirte the new built jar to the old one with the exact same name, like `cp org.eclipse.jdt.ui_3.33.200-SNAPSHOT.jar org.eclipse.jdt.ui_3.33.0.v20240822-0459.jar`.
